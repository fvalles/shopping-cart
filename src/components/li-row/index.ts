export { LiRow } from './li-row';
