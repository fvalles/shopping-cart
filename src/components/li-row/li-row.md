Basic list item row component:

```jsx
<LiRow />
```

List item row component with custom flex-flow and justify-content properties set:

```jsx
<LiRow flexFlow="row-reverse wrap" justifyContent="space-between" />
```
