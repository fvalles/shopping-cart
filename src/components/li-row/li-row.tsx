/**
 * Types
 */

import styled, { css } from 'styled-components';

interface LiRowProps {
  /** LiRow flex-flow property  */
  flexFlow?: string;
  /** LiRow justify-content property */
  justifyContent?: string;
}

/**
 * LiRow
 */

export const LiRow = styled.li<LiRowProps>`
  display: flex;
  ${({ justifyContent }) =>
    justifyContent &&
    css`
      justify-content: ${justifyContent};
    `}
  ${({ flexFlow }) =>
    flexFlow &&
    css`
      flex-flow: ${flexFlow};
    `}
`;
