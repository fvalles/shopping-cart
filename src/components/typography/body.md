Basic body text component with default color="body":

```jsx
<Body>Lorem ipsum</Body>
```

Body text component with color="primary":

```jsx
<Body color="primary">Lorem ipsum</Body>
```
