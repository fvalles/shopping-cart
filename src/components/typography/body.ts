import styled from 'styled-components';
import { TypographyProps } from './common';

export const Body = styled.p<TypographyProps>`
  color: ${({ color = 'body', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 13px;
  font-weight: 400;
  letter-spacing: -0.01em;
  line-height: 18px;
`;
