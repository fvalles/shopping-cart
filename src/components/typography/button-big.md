Basic button big text component with default color="primary":

```jsx
<ButtonBig>Lorem ipsum</ButtonBig>
```

Button big text component with color="summaryText":

```jsx
<ButtonBig color="summaryText">Lorem ipsum</ButtonBig>
```
