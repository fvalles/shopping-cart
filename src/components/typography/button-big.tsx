import styled from 'styled-components';
import { TypographyProps } from './common';

export const ButtonBig = styled.span<TypographyProps>`
  color: ${({ color = 'primary', theme }) => theme.Colors[color]};
  font-family: ${({ theme }) => theme.Fonts.regular}, 'Avenir Next', 'Segoe UI',
    sans-serif;
  font-size: 20px;
  font-weight: bold;
  line-height: 25px;
`;
