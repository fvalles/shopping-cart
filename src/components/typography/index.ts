export { ButtonBig } from './button-big';
export { ButtonSmall } from './button-small';
export { H1 } from './h1';
export { H2 } from './h2';
export { H3 } from './h3';
export { ListItemBig } from './list-item-big';
export { ListItemMedium } from './list-item-medium';
export { ListItemSmall } from './list-item-small';
export { Span } from './span';
export { Subtitle } from './subtitle';
export { Title } from './title';
