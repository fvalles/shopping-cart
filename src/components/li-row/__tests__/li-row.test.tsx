import React from 'react';
import { ThemeProvider } from '@core/theme/theme-provider';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { H1 } from '@components/typography';
import { LiRow } from '../li-row';

describe('li row component', () => {
  it('should render with default css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <LiRow>
            <H1>Some text</H1>
            <H1>Some other text</H1>
          </LiRow>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('display', 'flex');
    expect(result).toMatchSnapshot();
  });

  it('should render with custom justify-content and flex-flow css properties', () => {
    const result = renderer
      .create(
        <ThemeProvider>
          <LiRow justifyContent="center" flexFlow="row nowrap">
            <H1>Some text</H1>
            <H1>Some other text</H1>
          </LiRow>
        </ThemeProvider>,
      )
      .toJSON();

    expect(result).toHaveStyleRule('display', 'flex');
    expect(result).toHaveStyleRule('flex-flow', 'row nowrap');
    expect(result).toHaveStyleRule('justify-content', 'center');
    expect(result).toMatchSnapshot();
  });
});
