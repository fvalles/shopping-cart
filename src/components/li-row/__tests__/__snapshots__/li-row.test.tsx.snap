// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`li row component should render with custom justify-content and flex-flow css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Medium,'Avenir Next','Segoe UI', sans-serif;
  font-size: 18px;
  -webkit-letter-spacing: -0.18px;
  -moz-letter-spacing: -0.18px;
  -ms-letter-spacing: -0.18px;
  letter-spacing: -0.18px;
  line-height: 24px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
}

<li
  className="c0"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
  <h1
    className="c1"
  >
    Some other text
  </h1>
</li>
`;

exports[`li row component should render with default css properties 1`] = `
.c1 {
  color: #212240;
  font-family: Avenir Medium,'Avenir Next','Segoe UI', sans-serif;
  font-size: 18px;
  -webkit-letter-spacing: -0.18px;
  -moz-letter-spacing: -0.18px;
  -ms-letter-spacing: -0.18px;
  letter-spacing: -0.18px;
  line-height: 24px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<li
  className="c0"
>
  <h1
    className="c1"
  >
    Some text
  </h1>
  <h1
    className="c1"
  >
    Some other text
  </h1>
</li>
`;
